apps/teca_temporal_reduction.cpp

```diff
@@ -0,0 +1,269 @@
+    unpack->set_properties("unpack", opt_vals);
+    mav->set_properties("mav", opt_vals);
+    cf_writer->set_properties("cf_writer", opt_vals);
+
+    bool have_file = opt_vals.count("input_file");
+    bool have_regex = opt_vals.count("input_regex");
+    if ((have_file && have_regex) || !(have_file || have_regex))
+    {
+        if (mpi_man.get_comm_rank() == 0)
+        {
+            TECA_FATAL_ERROR("Extacly one of --input_file or --input_regex can be specified. "
+                "Use --input_file to activate the multi_cf_reader "
+                "and --input_regex to activate the cf_reader")
+        }
+        return -1;
+    }
+
+    if (!opt_vals["x_axis_variable"].defaulted())
+    {
+        cf_reader->set_x_axis_variable(opt_vals["x_axis_variable"].as<string>());
+        mcf_reader->set_x_axis_variable(opt_vals["x_axis_variable"].as<string>());
+    }
+
+    if (!opt_vals["y_axis_variable"].defaulted())
+    {
+        cf_reader->set_y_axis_variable(opt_vals["y_axis_variable"].as<string>());
+        mcf_reader->set_y_axis_variable(opt_vals["y_axis_variable"].as<string>());
+    }
+
+    if (!opt_vals["z_axis_variable"].defaulted())
+    {
+        cf_reader->set_z_axis_variable(opt_vals["z_axis_variable"].as<std::string>());
+        mcf_reader->set_z_axis_variable(opt_vals["z_axis_variable"].as<std::string>());
+    }
+
+    if (!opt_vals["t_axis_variable"].defaulted())
+    {
+        cf_reader->set_t_axis_variable(opt_vals["t_axis_variable"].as<std::string>());
+        mcf_reader->set_t_axis_variable(opt_vals["t_axis_variable"].as<std::string>());
+    }
+
+    if (opt_vals.count("input_file"))
+    {
+        mcf_reader->set_input_file(opt_vals["input_file"].as<string>());
+        vv_mask->set_input_connection(mcf_reader->get_output_port());
+    }
+    else if (opt_vals.count("input_regex"))
+    {
+        cf_reader->set_files_regex(opt_vals["input_regex"].as<string>());
+        vv_mask->set_input_connection(cf_reader->get_output_port());
+    }
+
+    unpack->set_input_connection(vv_mask->get_output_port());
+
+    mav->set_input_connection(unpack->get_output_port());
+    mav->set_thread_pool_size(1);
+    mav->set_stream_size(2);
+
+    if (!opt_vals["interval"].defaulted())
+    {
+        mav->set_interval(opt_vals["interval"].as<string>());
+    }
+
+    if (!opt_vals["operator"].defaulted())
+    {
+        mav->set_operator(opt_vals["operator"].as<string>());
+    }
+
+    if (opt_vals.count("fill_value"))
+    {
+        mav->set_fill_value(opt_vals["fill_value"].as<double>());
+    }
+
+    if (opt_vals["ignore_fill_value"].as<int>() == 0)
+    {
+        mav->set_use_fill_value(1);
+    }
+    else
+    {
+        mav->set_use_fill_value(0);
+    }
+
+    if (opt_vals.count("point_arrays"))
+        mav->set_point_arrays(opt_vals["point_arrays"].as<std::vector<std::string>>());
+
+    cf_writer->set_input_connection(mav->get_output_port());
+    cf_writer->set_thread_pool_size(1);
```

Comment:
This should be -1 and there should be a command line argument to modify it

Reply:
Do we need the two below or only one of them?

    if (!opt_vals["n_threads"].defaulted())
        mav->set_thread_pool_size(opt_vals["n_threads"].as<int>());
    else
        mav->set_thread_pool_size(-1);

    if (!opt_vals["n_threads"].defaulted())
        cf_writer->set_thread_pool_size(opt_vals["n_threads"].as<int>());
    else
        cf_writer->set_thread_pool_size(-1);